Update Runtime Provisioner documentation (#784)
kyma-incubator/compass: docs/provisioner/01-01-provisioner-overview.md

type: Overview
---

> The Runtime Provisioner is a Compass component responsible for provisioning, installing, and deprovisioning clusters with Kyma (Kyma Runtimes). The relationship between clusters and Runtimes is 1:1. The Runtime Provisioner is registered in the Director as an integration system. 

Reviewer: `Runtime Provisioner` vs. `Provisioner`? We need more consistency throughout the Compass docs. Also: `Director` vs. `the Director`?

It is powered by [Hydroform](https://github.com/kyma-incubator/hydroform) and it allows you to provision the clusters in the following ways:
- [through Gardener](08-02-provisioning-gardener.md) on:
    * GCP
    * Microsoft Azure
    * Amazon Web Services (AWS).
    

> Note that the operations of provisioning and deprovisioning are asynchronous. The operation of provisioning returns the Runtime Operation Status containing the runtime ID and the operation ID. The operation of deprovisioning returns the operation ID. You can use the operation ID to [check the Runtime Operation Status](08-03-runtime-operation-status.md).

Reviewer: ```suggestion
Note that the operations of provisioning and deprovisioning are asynchronous. Provisioning returns the Runtime Operation Status that contains the Runtime ID and the operation ID. Deprovisioning returns the operation ID. You can use the operation ID to [check the Runtime Operation Status](08-03-runtime-operation-status.md).
```

Reviewer: Consider putting that info in a **NOTE**.

Author: I didn't cos there are already other notes and on top of looking ugly that would probably disappear amongst them.

type: Overview
---

The Runtime Provisioner is a Compass component responsible for provisioning, installing, and deprovisioning clusters with Kyma (Kyma Runtimes). The relationship between clusters and Runtimes is 1:1. The Runtime Provisioner is registered in the Director as an Integration System. 

> It allows you to provision the clusters in the following ways:

Reviewer: ```suggestion
It allows you to provision clusters in the following ways:
```

- [through Gardener](08-02-provisioning-gardener.md) on:
    * GCP
    * Microsoft Azure
    * Amazon Web Services (AWS).
    

> During the operation of provisioning, you can pass a list of Kyma components you want installed on the provisioned Runtime with their custom configuration, as well as a custom Runtime configuration. 

Reviewer: ```suggestion
During the operation of provisioning, you can pass a list of Kyma components you want to install on the provisioned Runtime with their custom configuration, as well as a custom Runtime configuration. 
```

```bash
kubectl -n compass-system port-forward svc/compass-provisioner 3000:3000
```

> When making a call to the Runtime Provisioner, make sure to attach a Tenant header to the request.

Reviewer: ```suggestion
When making a call to the Runtime Provisioner, make sure to attach a tenant header to the request.
```

Reviewer: Why the capital letter for tenant? Also: `Runtime Provisioner` vs. `Provisioner`?
